File: 1013-fibonacci-number/1013-fibonacci-number.java
class Solution {
    public int print(int n,int []dp){
    if(n<=1){
        return n;
    }
        if(dp[n]!=-1){
            return dp[n];
        }
        return dp[n]=print(n-1,dp)+print(n-2,dp);
    }
    public int fib(int n) {
        int dp[]=new int[n+1];
        Arrays.fill(dp,-1);
        return print(n,dp);
    }
}

File: 1013-fibonacci-number/1013-fibonacci-number.cpp
class Solution {
public:
    int fib(int n) {
        if(n==0){
            return 0;
        }
        int prev=1;
        int pprev=1;
        int curr=prev;
        for(int i=2;i<n;i++){
            curr = prev+pprev;
            pprev = prev;
            prev = curr;
        }
        return curr;
    }
};